### ISSUE TITLE
[Feature] Publish images to alternative registry

### ISSUE THREAD

**Issue by anishasthana** [ray-project/kuberay#1133]
### Search before asking

- [X] I had searched in the [issues](https://github.com/ray-project/kuberay/issues) and found no similar feature requirement.

### Description

DockerHub allows you to pull only 100 images per 6 hour window[1], which isn't a lot. This results in the following error occurring fairly frequently.

```
Failed to pull image "kuberay/operator:v0.3.0": rpc error: code = Unknown desc = initializing source docker://kuberay/operator:v0.3.0: reading manifest v0.3.0 in docker.io/kuberay/operator: toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit
```

There are other container registries such as quay.io we could push images to instead (or as well) so as to get around the rate limit issues.

[1] https://docs.docker.com/docker-hub/download-rate-limit/

### Use case

Reduce instances of rate limit errors when pulling images from DockerHub.

### Related issues

https://github.com/ray-project/kuberay/issues/829
https://github.com/opendatahub-io/distributed-workloads/issues/47

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

**Comment by nvtkaszpir** [comment]
you can always make your own registry and pull/push images there.

**Comment by anishasthana** [comment]
Yep, that's what we're currently doing -- we just copied images over into our own registry. That being said, it's just another point of friction for new users, so in an ideal world we lower the bar to entry for anyone.

**Comment by kevin85421** [comment]
@anishasthana Do you have any recommendation for this issue? Thanks!

**Comment by Jeffwan** [comment]
@kevin85421 Can anyscale offer some premium plans for dockerhub kuberay?  or just host kuberay under dockerhub ray-project?

**Comment by kevin85421** [comment]
Thank you, @anishasthana, for the insightful suggestion! We will discuss this internally to determine if there is an existing solution in Ray or not, and we will get back to you as soon as possible.

**Comment by kevin85421** [comment]
This issue can be closed by #1307. Thank @anishasthana for the guidance! The next step is to update [the doc](https://github.com/ray-project/kuberay/blob/master/docs/deploy/docker.md) and may be change the default image registry to `Quay`.
